design/design-doc-observe-only-resources.md

```diff
@@ -0,0 +1,703 @@
+Crossplane without taking ownership of them.
+This would allow users to integrate existing cloud resources with the Crossplane
+ecosystem without giving full ownership.
+
+## Goals
+
+- Introduce a way to observe existing resources without taking ownership of them
+in Crossplane.
+- Enable seamless integration of existing cloud resources with the Crossplane
+ecosystem.
+
+### Non-goals
+
+- Partially managing a resource by observing a subset of fields.
+
+This may seem similar to the concept of observing resources, but there is a
+fundamental difference. In this scenario, we want to use certain parameters
+during the creation of the resource, whereas observing resources is intended to
+be a completely read-only operation that should never make any changes to the
+external system, including during the creation process.
+
+## Proposal
+
+We will introduce a new `managementPolicy` field in the spec of Managed
+Resources with `ObserveOnly` as one of the options. Additionally, we will change
+the API of the Managed Resources to include all the fields in the
+`spec.forProvider` under the `status.atProvider` to represent the full state of
+the resource on the external system. This will enable a clear separation between
+the desired state and the observed state of the resource and when the
+`managementPolicy` is set to `ObserveOnly`, only `status.atProvider` will be
+updated with the latest observation of the resource.
+
+### Management Policy
+
+To support observing resources without taking ownership, we will introduce a new
+spec named `managementPolicy` to the Managed Resources. We will also deprecate
+the existing `deletionPolicy` in favor of the new spec since they will be
+controlling the same behavior; that is, how should the changes on the CR affect
+the external cloud resource.
+
+This new policy will have the following values:
+
+- `FullControl`(Default): Crossplane will fully manage and control the external
+resource, including deletion when the CR is deleted
+(same as `deletionPolicy: Delete`).
+- `OrphanOnDelete`: Crossplane will orphan the external resource when the CR is
+deleted (same as `deletionPolicy: Orphan`).
+- `ObserveOnly`: Crossplane will only observe the external resource and will not
+make any changes or deletions.
+
+As indicated above, `FullControl` and `OrphanOnDelete` policies will behave
+precisely the same as the deletion policies we have today, including keeping the
+default behavior the same. We will introduce the new behavior with the
+`ObserveOnly` option, which would be pretty similar to what we have today to
+[import existing managed resources], but instead of starting to manage after
+import, we will not make any modifications to the external resource and only
+sync status back.
+
+```yaml
+apiVersion: ec2.aws.crossplane.io/v1beta1
+kind: VPC
+metadata:
+  annotations:
+    crossplane.io/external-name: vpc-12345678
+  name: observe-vpc
+spec:
+  managementPolicy: ObserveOnly
+  forProvider:
+    region: us-east-1
```

Comment:
It may be worth a small note pointing out that `spec.forProvider.region` is a bit special for OO resources. It's more an identifier than desired state.

Reply:
Good call, done: https://github.com/crossplane/crossplane/pull/3531/commits/d31602a09627bc44c10a33e48be5dbf08354126e